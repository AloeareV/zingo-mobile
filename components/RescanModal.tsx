/* eslint-disable react-native/no-inline-styles */
import React from 'react';
import {View, ScrollView, SafeAreaView, Image, Text} from 'react-native';
import {RegText, ZecAmount, UsdAmount, zecPrice} from './Components';
import Button from './Button';
import {useTheme} from '@react-navigation/native';

type RescanModalProps = {
  closeModal: () => void;
  birthday?: number;
  startRescan: () => void;
};

const RescanModal: React.FunctionComponent<RescanModalProps> = ({birthday, startRescan, closeModal, totalBalance}) => {
  const {colors} = useTheme();

  const doRescanAndClose = () => {
    startRescan();
    closeModal();
  };

  return (
    <SafeAreaView
      style={{
        display: 'flex',
        justifyContent: 'flex-start',
        alignItems: 'stretch',
        height: '100%',
        backgroundColor: colors.background,
      }}>
      <View
        style={{display: 'flex', alignItems: 'center', paddingBottom: 10, backgroundColor: colors.card, zIndex: -1, paddingTop: 10}}>
        <Image source={require('../assets/img/logobig-zingo.png')} style={{width: 80, height: 80, resizeMode: 'contain'}} />
        <ZecAmount size={36} amtZec={totalBalance.total} style={{opacity: 0.4}} />
        <RegText color={colors.money} style={{marginTop: 5, padding: 5}}>Rescan</RegText>
        <View style={{ width: '100%', height: 1, backgroundColor: colors.primary}}></View>
      </View>

      <ScrollView
        style={{maxHeight: '85%'}}
        contentContainerStyle={{
          flexDirection: 'column',
          alignItems: 'stretch',
          justifyContent: 'flex-start',
        }}>

        <View style={{display: 'flex', margin: 20}}>
          <RegText>
            This will re-fetch all transactions and rebuild your shielded wallet, starting from block height {birthday}.
            It might take several minutes.
          </RegText>
        </View>
      </ScrollView>

      <View style={{flexGrow: 1, flexDirection: 'row', justifyContent: 'center', alignItems: 'center', margin: 20}}>
        <Button type="Primary" title="Rescan" onPress={doRescanAndClose} />
        <Button type="Secondary" title="Close" style={{marginLeft: 10}} onPress={closeModal} />
      </View>
    </SafeAreaView>
  );
};

export default RescanModal;
