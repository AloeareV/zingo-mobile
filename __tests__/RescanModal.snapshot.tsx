/**
 * @format
 */

import 'react-native';
import React from 'react';

import { create } from 'react-test-renderer';
import RescanModal from '../components/RescanModal';

jest.useFakeTimers();

// test suite
describe('Component RescanModal - test', () => {
  //snapshot test
  test('RescanModal - snapshot', () => {
    const rescanModal = create(<RescanModal closeModal={() => {}} birthday={0} startRescan={() => {}} />);
    expect(rescanModal.toJSON()).toMatchSnapshot();
  });
});
