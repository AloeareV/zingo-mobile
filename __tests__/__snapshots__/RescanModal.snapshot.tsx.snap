// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component RescanModal - test RescanModal - snapshot 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "alignItems": "stretch",
      "backgroundColor": "rgb(242, 242, 242)",
      "display": "flex",
      "height": "100%",
      "justifyContent": "flex-start",
    }
  }
>
  <RCTScrollView
    contentContainerStyle={
      Object {
        "alignItems": "stretch",
        "flexDirection": "column",
        "justifyContent": "flex-start",
      }
    }
    style={
      Object {
        "maxHeight": "85%",
      }
    }
  >
    <View>
      <View>
        <View
          style={
            Object {
              "alignItems": "center",
              "backgroundColor": "rgb(255, 255, 255)",
              "paddingBottom": 25,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 28,
                "marginTop": 5,
                "padding": 5,
              }
            }
          >
            Rescan Wallet
          </Text>
        </View>
        <View
          style={
            Object {
              "alignItems": "center",
              "display": "flex",
              "marginTop": -25,
            }
          }
        >
          <Image
            source={
              Object {
                "testUri": "../../../assets/img/logobig.png",
              }
            }
            style={
              Object {
                "height": 50,
                "resizeMode": "contain",
                "width": 50,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "display": "flex",
            "margin": 10,
          }
        }
      >
        <Text
          style={
            Array [
              Object {
                "color": "rgb(28, 28, 30)",
                "fontSize": 18,
                "fontWeight": "600",
                "opacity": 0.87,
              },
            ]
          }
        >
          This will re-fetch all transactions and rebuild your shielded wallet, starting from block height 
          0
          . It might take several minutes.
        </Text>
      </View>
    </View>
  </RCTScrollView>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "flexGrow": 1,
        "justifyContent": "center",
        "margin": 20,
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "backgroundColor": "rgb(0, 122, 255)",
          "borderRadius": 10,
          "opacity": 1,
          "padding": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(242, 242, 242)",
            "fontWeight": "bold",
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Rescan
      </Text>
    </View>
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "borderColor": "rgb(28, 28, 30)",
          "borderRadius": 10,
          "borderWidth": 1,
          "marginLeft": 10,
          "opacity": 1,
          "padding": 10,
          "paddingLeft": 20,
          "paddingRight": 20,
          "width": "40%",
        }
      }
    >
      <Text
        style={
          Object {
            "color": "rgb(28, 28, 30)",
            "fontWeight": "bold",
            "textAlign": "center",
            "textTransform": "uppercase",
          }
        }
      >
        Close
      </Text>
    </View>
  </View>
</RCTSafeAreaView>
`;
